src/cognito/index.ts

```diff
@@ -2,19 +2,24 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { MapAuthHelper, SDKAuthHelper } from "../common/types";
-import { fromCognitoIdentityPool } from "@aws-sdk/credential-providers";
+import { FromCognitoIdentityPoolParameters, fromCognitoIdentityPool } from "@aws-sdk/credential-providers";
 import { Signer } from "@aws-amplify/core";
 import { AwsCredentialIdentity } from "@aws-sdk/types";
 /**
  * Creates an auth helper instance using credentials from Cognito.
  *
  * @param identityPoolId Cognito Identity Pool Id
  */
-export async function withIdentityPoolId(identityPoolId: string): Promise<MapAuthHelper & SDKAuthHelper> {
+export async function withIdentityPoolId(
+  identityPoolId: string,
+  options?: Partial<FromCognitoIdentityPoolParameters>,
+): Promise<MapAuthHelper & SDKAuthHelper> {
```

Comment:
It deserve a new function than extending withIdentityPoolId.

I would propose a new function like fromCognitoIdentityPool, and call `@aws-sdk/credential-providers`'s fromCognitoIdentityPool with the FromCognitoIdentityPoolParameters without modification. Since this change basically merges everything inside options and customer can supply identityPoolId and region inside FromCognitoIdentityPoolParameters

Reply:
> Since this change basically merges everything inside options and customer can supply identityPoolId and region inside FromCognitoIdentityPoolParameters

Yes, with the current implementation they could but the information provided in the first param `identityPoolId` (which is required) will be the one used to invoke the underlying `fromCognitoIdentityPool` method.

I can make the change you're suggesting but it will require more work and I don't see the point of providing an interface with two different methods name that are doing the same thing but providing different input. This can be done with an overload (which is what I've proposed in this PR).

I've added a test case that cover the scenario you mention

Reply:
Is there a use case that customer can use identity pool without identity pool id (for example, by specifying other parameters)? If there is no suce use case I am OK with overloading.

Reply:
no, that's not possible. The underlying method (`fromCognitoIdentityPool`) would require to provide `identityPoolId` in the options. Any other field is optional (which is when the overload comes handy).